### ISSUE TITLE
API refactoring; new error handling flow

### ISSUE THREAD

**Issue by servocoder** [servocoder/RichFilemanager#270]
Changes in the API for **v2.7.0**

1. `getfile` API method should be RENAMED to `getinfo` and return either file or folder stats. Restriction to handle files only should be removed.

2. `getfolder` API method should be RENAMED to `readfolder`. No changes in functionality.

3. API methods `readfile`, `getimage` and `download` should USE THE SAME FLOW to read file and write it to the output. The difference will be in headers and restrictions (validation checks) only.  Read/Write flow features which should be implemented (preferably):
    - read in small chunks in order to handle big size files;
    - handle HTTP RANGE for media files;

4. `editfile` API method should be REMOVED, instead of it the `readfile` is used at the client-side.

5. `download` API method should be simplified and handled with a single request. Implemented within the commit af60b300a79e4e8ec67c6d32842b45fcaa063e1b

6. The error handling. When fire an error you have to add the following HTTP headers to the response:

```php
        header('Content-Type: application/json');
        header('HTTP/1.1 500 Internal Server Error');
```

Also API error format sould correspond to JSON API standard. Imlemented within #210

**Comment by servocoder** [comment]
@fabriceci, @gmkll, @jsooter, @stevelittlefish, @jlaustill, @forestlake, @sinanbozkus, @richeflits 

I have reworked the API (located at the **dev** branch). Not great but very important changes to keep consistency and remove confusing things. Please, take a look the list and update the connectors that you are maintained.

@fabriceci, @gmkll, @jlaustill, @forestlake, @sinanbozkus, @richeflits  I have made some modifications trying to adjust connectors, but my exprience with non-PHP connectors is not enough to cover all the changes. I don't even have a chance to test the modifications I made, so look through your connectors thoroughly, test updated code and implement the missing features, please. (create PR to the **dev** branch)

@jsooter, @stevelittlefish Both Python Flask connectors are standalone packages, so I had no chance to modify them. Them will become outdated after the v2.7.0 release. Take care to update your connectors, please.

**Comment by servocoder** [comment]
7. New `seekfolder` API method, see #271

**Comment by servocoder** [comment]
8. Modified `attributes` list in the [API response](https://github.com/servocoder/RichFilemanager/wiki/API#response-structure).

Removed "timestamp" attribute. "created" and "modified" attributes now contain unix timestamps instead of formatted datetime strings. Datetime formatting is performed at the client-side now.

**Comment by servocoder** [comment]
Released in v2.7.0
